src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepositoryImplTest.java

```diff
@@ -135,6 +135,37 @@ void should_find_deleted_envelopes_by_status() {
         });
     }
 
+    @Test
+    void should_find_envelopes_by_container_and_file_name() {
+        // given
+        final String fileName = "foo.bar";
+        final String container = "bar";
+
+        // and
+        repo.insert(new NewEnvelope(container, fileName, now(), now(), Status.DISPATCHED));
+
+        // when
+        Optional<Envelope> result = repo.find(fileName, container);
+
+        // then
+        assertThat(result).hasValueSatisfying(envelope -> {
+            assertThat(envelope.fileName).isEqualTo(fileName);
+            assertThat(envelope.container).isEqualTo(container);
+        });
+    }
+
+    @Test
+    void should_return_empty_optional_when_envelope_for_given_container_and_file_name_does_not_exist() {
```

Comment:
Would be perfect if we checked that we get None when file name matches but container doesn't and when container matches but file name doesn't.